### ISSUE TITLE
findByIndex selector does not accept complete JSON string

### ISSUE THREAD

**Issue by ricellis** [cloudant/java-cloudant#137]
The `Database#findByIndex()` methods accept an incomplete JSON object for the string representing the selector:
`"selector" : {}`

It should accept a real JSON object as well e.g.:
`{"selector" : {} }`

It may be beneficial to have a `Selector` object that serializes to appropriate JSON.

**Comment by alkos333** [comment]
Selector object with supporting DSL for CQ would be a nice addition.  Is that what you are thinking, @ricellis?

**Comment by ricellis** [comment]
@alkos333 thanks for the feedback. I hadn't given a lot of thought to how the `Selector` object would be created, but my initial thought was fluent style method chaining, maybe something like:
`Selector.field("year").gt(2015)`
as that is something we have experimented with before.
A DSL could work out nicely too, but I wonder if it might be a step too far for this. Did you have something in mind?

**Comment by alkos333** [comment]
@ricellis, yes that is sufficient for the client.  I was thinking something along the lines of http://www.querydsl.com/ , but would surely be outside of the scope of the client here.

**Comment by ricellis** [comment]
I've raised issue #184 to cover the easier generation of selectors based on the discussion here about using a DSL/helper rather than a String.
The core issue here that `findByIndex()` only accepts JSON fragment strings was fixed as a side effect of the work for #101.
Some tests should be added to validate the fix from #101 and the `Database` documentation updated to use a complete JSON string, which is easier to obtain from serializing and object than the fragment.
